### PR TITLE
evaluation fcn/ test cases

### PR DESCRIPTION
**Pull request by c4n**
Not sure if this function is needed, since we already did CV.
By applying this fcn, it means that we are going to use out training set as a test set for X_set

In addition, classification_report() fcn only produces output as a text string... so I deleted the following variables:
-         self.precision = None
-        self.recall = None
-        self.f1 = None

### CODE REVIEW

**Comment by stevenzim** [comment]
Perhaps at a later time we can take the text string and then store them into the appropropriate attributes e.g. self.precision.
